Content.Server/Disease/DiseaseDiagnosisSystem.cs

```diff
@@ -0,0 +1,418 @@
+                return;
+            }
+            _popupSystem.PopupEntity(Loc.GetString("diagnoser-insert-swab", ("machine", uid), ("swab", args.Used)), uid, Filter.Entities(args.User));
+
+
+            machine.Disease = swab.Disease;
+            EntityManager.DeleteEntity(args.Used);
+
+            AddQueue.Enqueue(uid);
+            UpdateAppearance(uid, true, true);
+            SoundSystem.Play(Filter.Pvs(uid), "/Audio/Machines/diagnoser_printing.ogg", args.User);
+        }
+        /// <summary>
+        /// This handles the vaccinator machine up
+        /// until it's turned on. It has some slight
+        /// differences in checks from the diagnoser.
+        /// </summary>
+        private void OnAfterInteractUsingVaccine(EntityUid uid, DiseaseVaccineCreatorComponent component, AfterInteractUsingEvent args)
+        {
+            if (args.Handled || !args.CanReach)
+                return;
+
+            if (TryComp<ApcPowerReceiverComponent>(uid, out var power) && !power.Powered)
+                return;
+
+            if (!HasComp<HandsComponent>(args.User) || HasComp<ToolComponent>(args.Used)) //This check ensures tools don't break without yaml ordering jank
+                return;
+
+            if (!TryComp<DiseaseSwabComponent>(args.Used, out var swab) || swab.Disease == null || !swab.Disease.Infectious)
+            {
+                _popupSystem.PopupEntity(Loc.GetString("diagnoser-cant-use-swab", ("machine", uid), ("swab", args.Used)), uid, Filter.Entities(args.User));
+                return;
+            }
+            _popupSystem.PopupEntity(Loc.GetString("diagnoser-insert-swab", ("machine", uid), ("swab", args.Used)), uid, Filter.Entities(args.User));
+            var machine = Comp<DiseaseMachineComponent>(uid);
+            machine.Disease = swab.Disease;
+            EntityManager.DeleteEntity(args.Used);
+
+            AddQueue.Enqueue(uid);
+            UpdateAppearance(uid, true, true);
+        }
+
+        /// <summary>
+        /// This handles swab examination text
+        /// so you can tell if they are used or not.
+        /// </summary>
+        private void OnExamined(EntityUid uid, DiseaseSwabComponent swab, ExaminedEvent args)
+        {
+            if (args.IsInDetailsRange)
+            {
+                if (swab.Used)
+                    args.PushMarkup(Loc.GetString("swab-used"));
+                else
+                    args.PushMarkup(Loc.GetString("swab-unused"));
+            }
+        }
+
+        ///
+        /// Helper functions
+        ///
+
+        /// <summary>
+        /// This assembles a disease report
+        /// With its basic details and
+        /// specific cures (i.e. not spaceacillin).
+        /// The cure resist field tells you how
+        /// effective spaceacillin etc will be.
+        /// </summary>
+        private string AssembleDiseaseReport(DiseasePrototype disease)
+        {
+            string report = string.Empty;
```

Comment:
this should really be a `FormattedMessage` so you can use `PushMarkup` and `PushNewline` etc